hypertrace-gradle-publish-maven-central-plugin/src/main/java/org/hypertrace/gradle/publishing/PublishMavenCentralPlugin.java

```diff
@@ -156,9 +157,10 @@ private void addJavaLibraryPublicationWhenApplied(PublicationContainer publicati
 
           // scm
           String repoName = this.extension.repoName.get();
-          String scmConnection = String.format("scm:git:git://github.com/hypertrace/%s.git", repoName);
-          String scmDeveloperConnection = String.format("scm:git:ssh://github.com:hypertrace/%s.git", repoName);
-          String scmUrl = String.format("https://github.com/hypertrace/%s/tree/main", repoName);
+          String scmOrganization = this.extension.scmOrganization.get();
```

Comment:
existing bug on the line above but we should never explicitly resolve providers if we can help it (that is, don't call `get()`). The reason here is that we want evaluation to be lazy - as soon as this code executes, any later changes in the configuration (for these two properties) will be disregarded which may work on the default path, but is easily broken.

The general alternative which looks feasible here (because the pom attributes accept providers) is using `map` or `flatMap` to create new providers off of the input ones. It's a little ugly to merge two, but something like this should work:
```java
Provider<String> qualifiedRepoProvider =
    this.extension.scmOrganization.flatMap(
        repoName ->
            this.extension.repoName.map(
                orgName -> String.format("%s/%s", orgName, repoName)));
Provider<String> scmConnectionProvider =
    qualifiedRepoProvider.map(qualifiedRepo -> String.format("scm:git:git://github.com/%s.git", qualifiedRepo));
...
mavenPom.scm(
    mavenPomScm -> {
      mavenPomScm.getConnection().set(scmConnectionProvider);
      ...
```

Reply:
Makes sense! I can fix up both of them 👍 This is the kind of ugliness that is best encapsulated in a plugin so that the rest of the projects don't have to worry about running into problems with this plugin resolving providers 😃